pkg/db/test/db.go

```diff
@@ -112,3 +113,59 @@ func connectDB(name string) (db *sql.DB, err error) {
 
 	return sql.Open(cfg.DriverName, connStr+" password="+adminPassword)
 }
+
+// cleanupDB attempts to remove the test container created by EnsureDBReady
+func cleanupDB() {
+	cmd := exec.Command("docker", "rm", "-v", "-f", "go-base-postgres")
+	err := cmd.Run()
+	if err != nil {
+		fmt.Println("error during db cleanupdb: ", err)
+	}
+}
+
+// EnsureDBReady is a helper utility that can be inserted into a test during development to make it simpler
+// to run the test in isolation. Docker will be called to start a db that is read  to be used with GetDatabase.
+// Example usage:
+//
+//   err, cleanup := dbtest.EnsureDBReady(ctx)
+//   require.NoError(t, err)
+//   defer cleanup()
+//
+//   _, db := dbtest.GetDatabase(t)
+//   defer db.Close()
+//
+// While the code is somewhat robust to having existing dbs running, this is not intended to be left in the test code.
+// The goal is that you can run individual tests via your IDE integrations or using the CLI, e.g.
+//   go test -run ^TestRetentionHandler$`
+func EnsureDBReady(ctx context.Context) (error, func()) {
```

Comment:
I'm afraid it would cause a race condition when lots of tests will try to ensure the container at the same time (each test package is run in a separate process even). I think this is very dangerous to use it like this.

Reply:
it is only supposed to be used during development, per the docs string

>  this is not intended to be left in the test code. 

You can insert it into a test that requires a DB and it makes it just slightly easier to run via IDE integrations